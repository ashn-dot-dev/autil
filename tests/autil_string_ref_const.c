#define AUTIL_IMPLEMENTATION
#include "../autil.h"
#include "test.h"

int
main(void)
{
    struct autil_string const* const s = autil_string_new_cstr("foobar");

    ASSERT(*autil_string_ref_const(s, 0) == 'f');
    ASSERT(*autil_string_ref_const(s, 1) == 'o');
    ASSERT(*autil_string_ref_const(s, 2) == 'o');
    ASSERT(*autil_string_ref_const(s, 3) == 'b');
    ASSERT(*autil_string_ref_const(s, 4) == 'a');
    ASSERT(*autil_string_ref_const(s, 5) == 'r');

    // Check index out of bounds.
    autil_string_ref_const(s, 6);
}
