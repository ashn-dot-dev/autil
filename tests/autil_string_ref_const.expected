error: [autil_string_ref_const] Index out of bounds (6)
